client/src/routes/login/tabs/SignUp.tsx

```diff
@@ -40,41 +46,88 @@ function SignUp () {
 	
 	const [passwordsMatch, setPasswordsMatch] = React.useState(true);
 
+	const [errorText, setErrorText] = React.useState('');
+
+	const getLeeftijdsGroep = (birthYear:any) => {
+		const estimatedAge = new Date().getFullYear() - birthYear;
+
+		if(estimatedAge < 18) {
+			return 1;
+		}
+		if(estimatedAge < 50) {
+			return 2;
+		}
+		return 3;
+	};
+
 	const signUp = useCallback(async (e: React.FormEvent<HTMLFormElement>) => {
 		e.preventDefault();
 
 		if (password === confirmPassword) {
-			
-			// TODO sign up logica
 
+			const fullName = `${voornaam}${(tussenvoegsel) ? ' '+tussenvoegsel : ''} ${achternaam}`;
+			const leeftijdsGroep = getLeeftijdsGroep(geboorteDatum.$y);
+			await API('signup').Create({
+				naam : fullName,
+				leeftijdsGroep : leeftijdsGroep,
+				level : 1,
```

Comment:
we also have an enum for the levels, use that its in the user context. if you want you can make a new file in the global folder and add the enums there. its easier to keep track of what is going on